tests/cpp/unit-tests/test_unit_cpp_client.py

```diff
@@ -67,25 +67,31 @@ def execute_cmd(cmd_list):
         cmd_list, stderr=PIPE, stdout=PIPE, stdin=PIPE, cwd=run_path)
     try:
         out, err = proc.communicate(timeout=timeout_limit)
+        print("OUTPUT:", out.decode("utf-8") if out else "None")
+        print("ERROR:", err.decode("utf-8") if err else "None")
+        if (proc.returncode != 0):
+            print("Return code:", proc.returncode)
         assert(proc.returncode == 0)
-        if out:
-            print("OUTPUT:", out.decode("utf-8"))
-        if err:
-            print("ERROR:", err.decode("utf-8"))
     except UnicodeDecodeError:
         output, errs = proc.communicate()
-        print("ERROR:", errs.decode("utf-8"))
+        print("OUTPUT:", out)
+        print("ERROR:", errs)
         assert(False)
     except TimeoutExpired:
         proc.kill()
         output, errs = proc.communicate()
         print(f"TIMEOUT: test timed out after test timeout limit of {timeout_limit} seconds")
         print("OUTPUT:", output.decode("utf-8"))
         print("ERROR:", errs.decode("utf-8"))
+        print("OUTPUT:", out)
```

Comment:
There have been cases where we don't get any output at all because of binary data in the print stream that can't be decoded. This will make sure that we get a minimum level of output in every case.